.github/workflows/pre-merge-e2e-test.yml

```diff
@@ -0,0 +1,43 @@
+name: Pre Merge E2E test
+
+on:
+  pull_request:
+    branches:
+      - main
+    types: [opened, reopened, synchronize]
+
+jobs:
+  e2e-tests:
+    name: E2e-tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 'stable'
+      - uses: azure/setup-kubectl@v3
+        with:
+          version: 'v1.26.1'
+      - name: Install oc
+        uses: redhat-actions/oc-installer@v1
+        with:
+          oc_version: '4.6'
+      - name: Login to cluster
+        env:
+          USERNAME: ${{ secrets.username }}
+          PASSWORD: ${{ secrets.password }}
+          API_ENDPOINT: ${{ secrets.api_endpoint }}
+        run: oc login "$API_ENDPOINT" -u "$USERNAME" -p "$PASSWORD" --insecure-skip-tls-verify
```

Comment:
I'd like us to not be coupled to `oc` and instead use `kubectl` directly, should figure out a different way to login to the cluster